File: ruby/predict.rb
require 'net/http'
require 'net/https'
require 'uri'
require 'json'
require 'dotenv'

Dotenv.load('../common/.env')

PREDICTION_SERVER = ENV['PREDICTION_SERVER']
DEPLOYMENT_ID = ENV['DEPLOYMENT_ID']
PREDICTION_ENDPOINT = PREDICTION_SERVER + "/predApi/v1.0/deployments/" + DEPLOYMENT_ID + "/predictions"
API_KEY = ENV['API_KEY']
DATAROBOT_KEY = ENV['DATAROBOT_KEY']

PAYLOAD = File.read('../common/payload_to_predict.json')


headers = {
    'Content-Type': 'application/json',
    'charset': 'UTF-8', 
    'datarobot-key': DATAROBOT_KEY, 
    'Authorization': 'Bearer ' + API_KEY
}

uri = URI.parse(PREDICTION_ENDPOINT)
https = Net::HTTP.new(uri.host,uri.port)
https.use_ssl = true

request = Net::HTTP::Post.new(uri.path)

request['Content-Type'] = 'application/json'
request['datarobot-key'] = DATAROBOT_KEY
request['Authorization'] = 'Bearer '+API_KEY

request.body = PAYLOAD

response = https.request(request)

puts response.code
puts response.body

File: node/predict.js
const got = require('got')
const fs = require('fs')
require('dotenv').config({ path: '../common/.env' })

const PREDICTION_SERVER = process.env.PREDICTION_SERVER
const DEPLOYMENT_ID = process.env.DEPLOYMENT_ID
const PREDICTION_ENDPOINT = `${PREDICTION_SERVER}/predApi/v1.0/deployments/${DEPLOYMENT_ID}/predictions`
const API_KEY = process.env.API_KEY
const DATAROBOT_KEY = process.env.DATAROBOT_KEY

async function makePrediction(){

    let payload_to_predict = JSON.parse(fs.readFileSync('../common/payload_to_predict.json').toString())

    let { statusCode, body } = await got.post(PREDICTION_ENDPOINT, {
        headers: {
            "Authorization": `Bearer ${API_KEY}`,
            "datarobot-key": DATAROBOT_KEY
        },
        json: payload_to_predict,
        responseType: 'json'
    })

    console.log(`Response status: ${statusCode}`)
    console.log(body)
}

makePrediction()
